### PR TITLE
Get rid of pre 1.6 reflection

### PR DESCRIPTION
**Pull request by RoiEXLab**
Addition to #1102
This should be all cross java reflection code in java...

I don't think all of this code is necessary, but I thought the sam thing about the MAC adress search - it prove me wrong...
I will remove this class entirely in the next PR, which requires us to do some testing...

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/1108)

### CODE REVIEW

File: src/games/strategy/net/DesktopUtilityBrowserLauncher.java [hunk 1]
```diff
@@ -32,24 +37,23 @@
...
+      }
+    }
+    if (failed) {
       // we use "toString()" or to "toASCIIString()" because "getPath()" sometimes does not work.
       final String url = uri.toString();
       final String osName = System.getProperty("os.name");
       try {
         if (osName != null && osName.startsWith("Mac OS")) {
-          try {
-            Class.forName("com.apple.eio.FileManager").getDeclaredMethod("open", String.class)
```
Comment (RoiEXLab) [inline src/games/strategy/net/DesktopUtilityBrowserLauncher.java]: This method doesn't even exist

File: src/games/strategy/net/DesktopUtilityBrowserLauncher.java [hunk 2]
```diff
@@ -79,18 +83,19 @@ public static void openFile(final File file) {
...
+        failed = false;
+      } catch (IOException e) {
+        ClientLogger.logError(e);
+      }
+    }
+    if (failed) {
       final String osName = System.getProperty("os.name");
       try {
-        if (osName != null && osName.startsWith("Mac OS")) {
-          Class.forName("com.apple.eio.FileManager").getDeclaredMethod("openURL", String.class)
```
Comment (RoiEXLab) [inline src/games/strategy/net/DesktopUtilityBrowserLauncher.java]: OpenURL method simply throws a RuntimeException. At least thats what the java decompiler tells me

File: src/games/strategy/net/DesktopUtilityBrowserLauncher.java [hunk 3]
```diff
@@ -79,18 +83,19 @@ public static void openFile(final File file) {
...
-    try { // attempt to use Desktop library from JDK 1.6+
-      final Class<?> d = Class.forName("java.awt.Desktop");
-      d.getDeclaredMethod("browse", URI.class)
-          .invoke(d.getDeclaredMethod("getDesktop").invoke(null), URI.create(url));
-      // above code mimicks: java.awt.Desktop.getDesktop().browse()
-    } catch (final Exception ignore) { // library not available or failed
+    boolean failed = true;// TODO check if the failed thing is necessary
+    if (Desktop.isDesktopSupported()) {
+      try {
+        Desktop.getDesktop().browse(URI.create(url));
```
Comment (DanVanAtta) [inline src/games/strategy/net/DesktopUtilityBrowserLauncher.java]: This block of code is familiar. Seems like you could extract it to a function that returns a boolean, and then set the value of the `failed` variable with it, ie:  `boolean failed = isDesktopSupport(..)` 
Probably would make the failed var unnecessary at this level even, so `if(isDesktopSupported(..))`

Comment (RoiEXLab) [inline src/games/strategy/net/DesktopUtilityBrowserLauncher.java]: As you may noticed, I have anothe PR in queue which removes most of the content of this class... Although it needs some testing it's very likely to replace this changes on merge...

**Pull request merged**